keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,842 @@
+  DeletionPolicy DeletionPolicy
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string
 }
 ```
 
-## ProvisonerDeleteBucket
+## BucketAccess
 
-This call is made to delete the bucket in the backend. If the bucket has already been deleted, then no error should be returned. The call to *ProvisonerDeleteBucket* MUST be idempotent.
+A resource to access a Bucket. BucketAccess is namespace-scoped
 
+```yaml
+BucketAccess {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketClaimSpec {
+    // BucketClaimName is the name of the BucketClaim.
+    // Exactly one of BucketClaimName or BucketName must be set.
+    // +optional
+    BucketClaimName string
+
+    // BucketName is the name of the Bucket for which
+    // credetials need to be generated
+    // Exactly one of BucketClaimName or BucketName must be set.
+    // +optional
+    BucketName string
+
+    // BucketAccessClassName is the name of the BucketAccessClass
+    BucketAccessClassName string
+
+    // CredentialsSecretName is the name of the secret that COSI should populate
+    // with the credentials. If a secret by this name already exists, then it is
+    // assumed that credentials have already been generated. It is not overridden.
+    CredentialsSecretName string
+	
+    // ServiceAccountName is the name of the serviceAccount that COSI will map
+	// to the OSP service account when IAM styled authentication is specified
+    CredentialsSecretName string	
+  }
+
+  Status BucketAccessStatus {
+	// AccessGranted indicates the successful grant of privileges to access the bucket
+	AccessGranted bool
+	
+    // AccountID is the unique ID for the account in the OSP
+    AccountID string
+  }
 ```
-message ProvisionerDeleteBucketRequest {
-    // This field is REQUIRED
-    string bucket_name = 1;
-    
-    map<string,string> bucket_context = 2;    
+
+## BucketAccessClass
+
+Resouce for configuring common properties for multiple BucketClaims. BucketAccessClass is a clustered resource
+
+```yaml
+BucketAccessClass {
+  TypeMeta
+  ObjectMeta
+
+  // AuthenticationType denotes the style of authentication
+  // It can be one of
+  // KEY - access, secret tokens based authentication
+  // IAM - implicit authentication of pods to the OSP based on service account mappings
+  AuthenticationType AuthenticationType
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string
 }
+```
+
+## BucketInfo
+
+Resource mounted into pods containing information for applications to gain access to buckets.
 
-message ProvisionerDeleteBucketResponse {
-     // Intentionally left blank
+```yaml
+BucketInfo {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketInfoSpec {
+    // BucketName is the name of the Bucket 
+    BucketName string
+
+	// AuthenticationType denotes the style of authentication
```

Comment:
In this struct, it seems there is a mix of spaces and tabs. Github is flagging the tabs with dark red bars from what I see. These seem to also be present elsewhere as well.

Reply:
untabified the entire document. It should look better now